### ISSUE TITLE
I just updated to 1.1.1, unity3d can't connect to socket.io server.

### ISSUE THREAD

**Issue by gzlock** [nhnent/socket.io-client-unity3d#15]
**ENV:**
Node.JS: 8.2.1
Node.JS socket.io: 2.0.3
socket.io-client-unity3d: 1.1.1 (download form github release page)
Unity3d: 2017.1.0 f3

----------

Server side i don't change anything.
Node.JS  socket.io client can connect socket.io server.
But unity3d 2017.1 + socket.io-client-unity3d 1.1.1 can't connect to server
In this case how to find the solution?

unity3d client code:
```c#
Socket io = Socket.Connect ("http://127.0.0.1:7001/game");
```

throw two errors:

```
socket.io => socket.io - http://localhost:7001/game got WWW error: 
UnityEngine.Debug:LogErrorFormat(String, Object[])
socket.io.SocketManager:<Update>m__2(Exception) (at Assets/Plugins/socket.io/SocketManager.cs:129)
UniRx.Operators.DoOnError:OnError(Exception) (at Assets/Plugins/UniRx/Scripts/Operators/Do.cs:212)
UniRx.Operators.Timeout:OnError(Exception) (at Assets/Plugins/UniRx/Scripts/Operators/Timeout.cs:110)
UniRx.Operators.FromCoroutine:OnError(Exception) (at Assets/Plugins/UniRx/Scripts/UnityEngineBridge/Operators/FromCoroutine.cs:55)
socket.io.<InitCore>c__Iterator0:MoveNext() (at Assets/Plugins/socket.io/SocketInitializer.cs:131)
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)

```

```
WWWErrorException: Exception of type 'UniRx.WWWErrorException' was thrown.
UniRx.InternalUtil.ThrowObserver`1[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/InternalUtil/ListObserver.cs:102)
UniRx.Operators.DoOnCompletedObservable`1+DoOnCompleted[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/Operators/Do.cs:272)
UniRx.Operators.DoOnErrorObservable`1+DoOnError[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/Operators/Do.cs:222)
UniRx.Operators.TimeoutObservable`1+Timeout[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/Operators/Timeout.cs:110)
UniRx.Operators.FromCoroutineObservable`1+FromCoroutine[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/UnityEngineBridge/Operators/FromCoroutine.cs:55)
socket.io.SocketInitializer+<InitCore>c__Iterator0.MoveNext () (at Assets/Plugins/socket.io/SocketInitializer.cs:131)
UnityEngine.SetupCoroutine.InvokeMoveNext (IEnumerator enumerator, IntPtr returnValueAddress) (at /Users/builduser/buildslave/unity/build/Runtime/Export/Coroutines.cs:17)
```

----------------

node.js client code:
```javascript
const io = require('socket.io-client');
const socket = io('http://127.0.0.1:7001/game');

socket.on('connect', () => {
  console.log('connect!');
});
```
![image](https://user-images.githubusercontent.com/2311740/28472500-610a7b3a-6e73-11e7-8a26-67527fb91afa.png)

**Comment by gzlock** [comment]
I downgraded the socket.io to v1.7.4, but still failed to connect
```
socket.io => socket.io - http://127.0.0.1:7001/game got WWW error: 
UnityEngine.Debug:LogErrorFormat(String, Object[])
socket.io.SocketManager:<Update>m__2(Exception) (at Assets/Plugins/socket.io/SocketManager.cs:129)
UniRx.Operators.DoOnError:OnError(Exception) (at Assets/Plugins/UniRx/Scripts/Operators/Do.cs:212)
UniRx.Operators.Timeout:OnError(Exception) (at Assets/Plugins/UniRx/Scripts/Operators/Timeout.cs:110)
UniRx.Operators.FromCoroutine:OnError(Exception) (at Assets/Plugins/UniRx/Scripts/UnityEngineBridge/Operators/FromCoroutine.cs:55)
socket.io.<InitCore>c__Iterator0:MoveNext() (at Assets/Plugins/socket.io/SocketInitializer.cs:131)
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)
```

```
WWWErrorException: Exception of type 'UniRx.WWWErrorException' was thrown.
UniRx.InternalUtil.ThrowObserver`1[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/InternalUtil/ListObserver.cs:102)
UniRx.Operators.DoOnCompletedObservable`1+DoOnCompleted[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/Operators/Do.cs:272)
UniRx.Operators.DoOnErrorObservable`1+DoOnError[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/Operators/Do.cs:222)
UniRx.Operators.TimeoutObservable`1+Timeout[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/Operators/Timeout.cs:110)
UniRx.Operators.FromCoroutineObservable`1+FromCoroutine[socket.io.Socket].OnError (System.Exception error) (at Assets/Plugins/UniRx/Scripts/UnityEngineBridge/Operators/FromCoroutine.cs:55)
socket.io.SocketInitializer+<InitCore>c__Iterator0.MoveNext () (at Assets/Plugins/socket.io/SocketInitializer.cs:131)
UnityEngine.SetupCoroutine.InvokeMoveNext (IEnumerator enumerator, IntPtr returnValueAddress) (at /Users/builduser/buildslave/unity/build/Runtime/Export/Coroutines.cs:17)

```

**Comment by ppz0th** [comment]
@gzlock 
In the stack trace, there is an Timeout observable which means that connection trial is timed out. This usually happens when the server and the client network setting is wrong. Or it's just my guessing that you working on Unity 2017 and I'm not sure our socket.io supports it;; We have not test under Unity 2017 yet. 

Anyway I'll check this issue. Can you give me the sample server code which you can connect to now?

**Comment by ppz0th** [comment]
@gzlock 
Hello~ I've just test with my own sample code below

* Server
```
var app = require('http').createServer(handler)
var fs = require('fs');

app.listen(7001);

function handler (req, res) {
  fs.readFile(__dirname + '/index.html',
  function (err, data) {
      if (err) {
          res.writeHead(500);
          return res.end('Error loading index.html');
      }

      res.writeHead(200);
      res.end(data);
  });
}

// Start socket.io
var io = require('socket.io')(app);
  
// game nsp
var game = io
  .of('/game')
  .on('connection', function (socket) {
    console.log('game connected~')
  });
```

* Client
```
using UnityEngine;
using socket.io;

namespace Sample {

    public class Namespace : MonoBehaviour {

        void Start() {
            var io = Socket.Connect("http://127.0.0.1:7001/game");

            io.On(SystemEvents.connect, () => {
                Debug.Log("Hello~, world.");
            });
        }

    }

}
```

I got no error;; Maybe it's Unity3d 2017 problem??

**Comment by gzlock** [comment]
yep, I just downgraded unity3d to 5.6.2, test no error.
But my project someone plugin downgraded fail, it has always throw an error:
```
TypeLoadException: Could not load type 'UnityEngine.ImageConversion' from assembly 'BehaviorDesignerEditor'.
```
Hope to support unity3d 2017.1, fighting!
👍

**Comment by HectorPulido** [comment]
Same problem, Works ok on 5.6.2 but It brokes on 2017.1 :c

**Comment by ppz0th** [comment]
@gzlock @HectorPulido 
Hello~ v1.1.2 just has been released and you can work on Unity3d 2017.1 now~
